addons/pos_restaurant/static/src/overrides/components/product_screen/actionpad_widget/actionpad_widget.js

```diff
@@ -49,19 +49,17 @@ patch(ActionpadWidget.prototype, {
         const orderChange = this.currentOrder.getOrderChanges().orderlines;
 
         const categories = Object.values(orderChange).reduce((acc, curr) => {
-            const categories =
-                this.pos.models["product.product"].get(curr.product_id)?.pos_categ_ids || [];
+            const category =
+                this.pos.models["product.product"].get(curr.product_id)?.pos_categ_ids[0] || [];
 
-            for (const category of categories) {
-                if (category) {
-                    if (!acc[category.id]) {
-                        acc[category.id] = {
-                            count: curr.quantity,
-                            name: category.name,
-                        };
-                    } else {
-                        acc[category.id].count += curr.quantity;
-                    }
+            if (category) {
```

Comment:
This `if` is useless as if `this.pos.models["product.product"].get(curr.product_id)?.pos_categ_ids[0]` returns a falsy value, it will be replaced by `[]` which is evaluated as a true statement in an `if`. (Not saying it is your fault as it was already there but can you remove it)

Reply:
You can actually keep the `if` but remove the `|| []` on the line above then

Reply:
Hi @adgu-odoo ,
Yes, it's useless thank you for the suggestion i have changed it accordingly.